docs/nipoppy/glossary.md

```diff
@@ -0,0 +1,68 @@
+## Glossary
+
+Here are some definitions for important/recurring terms used in the Nipoppy framework, in alphabetical order.
+
+### `bids_id`
+
+**Appears in**: `doughnut.csv`
+
+:   A BIDS-compliant participant identifier. Obtained by adding the `"sub-"` prefix to the [`dicom_id`](#dicom_id), which itself is derived from the [`participant_id`](#participant_id). A participant's raw BIDS data and derived imaging data are stored in directories named after their `bids_id`.
+
+### `datatype`
+
+**Appears in**: `manifest.csv`
+
+:   A BIDS-compliant "data type" value (see the [BIDS specification website](https://bids-specification.readthedocs.io/en/stable/common-principles.html#definitions) for a comprehensive list). The most common data types for magnetic resonance imaging (MRI) data are `"anat"`, `"func"`, and `"dwi"`.
+
+### `dicom_id`
+
+**Appears in**: `doughnut.csv`
+
+:   The [`participant_id`](#participant_id), stripped of any non-alphanumerical character. For studies that do not use non-alphanumerical characters in their participant IDs, this is exactly the same as [`participant_id`](#participant_id).
+
+### `participant_dicom_dir`
+
+**Appears in**: `doughnut.csv`
+
+:   The name of the directory in which the raw DICOM data (before the DICOM organization step) are found. Usually, this is the same as [`participant_id`](#participant_id), but depending on the study it could be different.
+
+### `participant_id`
+
+**Used interchangeably with**: `subject_id`
```

Comment:
Apparently yes: https://github.com/search?q=repo%3Aneurodatascience%2Fnipoppy+subject_id&type=code

Mostly in sample Slurm/SGE scripts, but also in the Freesurfer extractor and `catalog.py`. Also some things refer to "subject" instead of "participant". But you're right that it's not a command-line argument. We might still want to keep this though